src/legacy/server/saved_objects/saved_objects_mixin.js

```diff
@@ -41,10 +41,28 @@ import {
   createLogLegacyImportRoute,
 } from './routes';
 
+function getImportableAndExportableTypes({ kbnServer, visibleTypes }) {
+  const { savedObjectSchemas = {} } = kbnServer.uiExports;
+  return visibleTypes.filter(
+    type =>
+      !savedObjectSchemas[type] || savedObjectSchemas[type].isImportableAndExportable !== false
+  );
+}
+
 export function savedObjectsMixin(kbnServer, server) {
   const migrator = new KibanaMigrator({ kbnServer });
+  const mappings = migrator.getActiveMappings();
+  const allTypes = Object.keys(getRootPropertiesObjects(mappings));
+  const schema = new SavedObjectsSchema(kbnServer.uiExports.savedObjectSchemas);
+  const visibleTypes = allTypes.filter(type => !schema.isHiddenType(type));
+  const importableAndExportableTypes = getImportableAndExportableTypes({ kbnServer, visibleTypes });
 
   server.decorate('server', 'kibanaMigrator', migrator);
+  server.decorate(
+    'server',
+    'getSavedObjectsSchema',
+    () => new SavedObjectsSchema(kbnServer.uiExports.savedObjectSchemas)
+  );
```

Comment:
Can we add this to the Server interface in [`kbn_server.d.ts`](https://github.com/elastic/kibana/blob/master/src/legacy/server/kbn_server.d.ts#L53)?